### ISSUE TITLE
MethodError when using EvoTreeClassifier with MLJTuningTreeParzen

### ISSUE THREAD

**Issue by svilupp** [Evovest/EvoTrees.jl#216]
Behavior: When running the [MLJ tuning example](https://github.com/IQVIA-ML/TreeParzen.jl/blob/master/docs/examples/simple_mlj_demo/simple_mlj_demo.md) with EvoTreeClassifier, tuning fails due to a MethodError that ‘ EvoTreeClassifier{Softmax,Float32}(;…)’ doesn’t exist.

Root cause: I believe the issue comes from [this line](https://github.com/IQVIA-ML/TreeParzen.jl/blob/15224294dce354ce2148bea4e37066c2afa44654/src/MLJTreeParzen.jl#L266), which new hyperparams to a kwarg constructor extracted as ‘typeof(model)(; new_params…)’

Unfortunately, that requires a constructor ‘ EvoTreeClassifier{Softmax,Float32}(;…)’, but we have only the [generic one defined](https://github.com/Evovest/EvoTrees.jl/blob/2ff0f5a7a1ab7e85326212cf9c9bd48ead43cc50/src/models.jl#L194).

It can be solved by adding new kwarg constructors for more specific parametric types (eg, with ‘where {L,T}’).

Impact: I believe this impacts all Evo models when used with MLJTuningTreeParzen.

Proposal: I’m out until next week, but I’m happy to open the PR when I’m back.

For completeness, this happens on the latest version of EvoTrees. Julia 1.8.5., arm-based.

**Comment by jeremiedb** [comment]
Thanks for reporting this. If you can open the PR, that would be welcomed!

**Comment by ablaom** [comment]
I think the indicated line in TreeParzen.jl is quite brittle. I can imaging it failing for other MLJ models that don't have parameter-type-specific constructors. 

I'll open an issue at TreeParzen.jl. Unfortunately, this pkg has only sporadic maintenance. So you might want to go ahead with the local fix, which sounds innocuous enough.

**Comment by ablaom** [comment]
@svilupp Thanks for reporting, by the way.
